bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/FeatureInstaller.java

```diff
@@ -106,14 +116,34 @@ protected void activate(final Map<String, Object> config) {
         setOnlineRepoUrl();
         setLegacyRepoUrl();
         modified(config);
+        scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
I would maybe initialize where I declare it, for me that feels safer.

Reply:
No, because it needs to be (re-)initialized upon activation and shut down on deactivation.